src/solvers/flattening/boolbv_overflow.cpp

```diff
@@ -96,6 +95,43 @@ literalt boolbvt::convert_overflow(const exprt &expr)
       return !prop.lor(all_one, all_zero);
     }
   }
+  else if(expr.id()==ID_overflow_shl)
+  {
+    if(operands.size()!=2)
+      throw "operator "+expr.id_string()+" takes two operands";
+
+    const bvt &bv0=convert_bv(operands[0]);
+    const bvt &bv1=convert_bv(operands[1]);
+
+    std::size_t old_size=bv0.size();
+    std::size_t new_size=old_size*2;
+
+    bvt bv_ext=bv_utils.zero_extension(bv0, new_size);
+
+    bvt result=bv_utils.shift(bv_ext, bv_utilst::shiftt::LEFT, bv1);
+
+    // an undefined shift of a non-zero value always results in overflow
+    literalt undef=
+      bv_utils.rel(
+        bv1,
+        ID_gt,
+        bv_utils.build_constant(old_size, bv1.size()),
+        bv_utilst::representationt::UNSIGNED);
+
```

Comment:
I believe that the right-hand-side operand may be either signed or unsigned. I think the above will only work if the right-hand side has enough bits (i.e., sign extension may be needed).

Reply:
I was thinking that negative shifts should not be caught here, but indeed I need to make sure that this case is truly ignored.

Reply:
I've added this check in the updated revision.